File: Numerical_Methods_for_Physics_Garcia/edition2/Cpp/orthog.cpp
// orthog - Program to test if a pair of vectors 
// is orthogonal.  Assumes vectors are in 3D space
#include <iostream.h>

void main() {

  //* Initialize the vectors a and b
  double a[3+1], b[3+1];
  cout << "Enter the first vector" << endl;
  int i;
  for( i=1; i<=3; i++ ) {
    cout << "  a[" << i << "] = ";
    cin >> a[i];
  }
  cout << "Enter the second vector" << endl;
  for( i=1; i<=3; i++ ) {
    cout << "  b[" << i << "] = ";
    cin >> b[i];
  }

  //* Evaluate the dot product as sum over products of elements
  double a_dot_b = 0.0;
  for( i=1; i<=3; i++ )
    a_dot_b += a[i]*b[i];


  //* Print dot product and state whether vectors are orthogonal
  if( a_dot_b == 0.0 )
    cout << "Vectors are orthogonal" << endl;
  else {
    cout << "Vectors are NOT orthogonal" << endl;
    cout << "Dot product = " << a_dot_b << endl;
  }
}

File: Numerical_Methods_for_Physics_Garcia/edition2/Fortran/orthog.f
! orthog - Program to test if a pair of vectors
! is orthogonal.  Assumes vectors are in 3D space

      program orthog

      !* Initialize the vectors a and b
      integer*4 i
      real*8 a(3), b(3), a_dot_b
      write(*,*) 'Enter the first vector'
      do i=1,3
        write(*,*) '  a(', i, ') = '
        read(*,*) a(i)
      enddo
      write(*,*) 'Enter the second vector'
      do i=1,3
        write(*,*) '  b(', i, ') = '
        read(*,*) b(i)
      enddo

      !* Evaluate the dot product as sum over products of elements
      a_dot_b = 0.0
      do i=1,3
        a_dot_b = a_dot_b + a(i)*b(i)
      enddo

      !* Print dot product and state whether vectors are orthogonal
      if( a_dot_b .eq. 0.0 ) then
        write(*,*) 'Vectors are orthogonal'
      else
        write(*,*) 'Vectors are NOT orthogonal'
        write(*,*) 'Dot product = ', a_dot_b
      endif
      stop
      end
